src/Events.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core;
+
+/**
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class Events
+{
+    public const QUERY_PARAMETER_VALIDATE = 'api_platform.query_parameter_validate';
+    public const FORMAT_ADD = 'api_platform.format_add';
+
+    public const PRE_READ = 'api_platform.pre_read';
+    public const READ = 'api_platform.read';
```

Comment:
Couldn't we only have the PRE ad POST events?

Reply:
I was thinking that `PRE` and `POST` events were for the user whereas the other ones were for our internal use only.